### PR TITLE
Add New Port SATAN LOVES CAKE

### PR DESCRIPTION
**Pull request by MadShmupper**
Add New Port SATAN LOVES CAKE

# New Port for { SATAN LOVES CAKE }

## Game Information
- **Title**: { SATAN LOVES CAKE }
- **URL**: { https://baku.itch.io/satan-loves-cake }

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS

### Resolution Tests
Test all major resolutions:
- [x] 480x320
- [x] 640x480
- [ ] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

File: ports/satlovecake/SATAN LOVES CAKE.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+# PortMaster info
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
```
Comment (cdeletre) [inline ports/satlovecake/SATAN LOVES CAKE.sh:19]: Mod files sourcing is missing and `export PORT_32BIT="Y"` must be set before.
Sourcing `device_info.txt` is not needed anymore.
```suggestion
export PORT_32BIT="Y"
[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
```

File: ports/satlovecake/SATAN LOVES CAKE.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+# Enable logging
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# Set necessary exports
+export LD_LIBRARY_PATH="/usr/lib:/usr/lib32:$GAMEDIR/lib:$LD_LIBRARY_PATH"
+export GMLOADER_DEPTH_DISABLE=1
+export GMLOADER_SAVEDIR="$GAMEDIR/"
+export GMLOADER_PLATFORM="os_windows"
+export PORT_32BIT="Y"
```
Comment (cdeletre) [inline ports/satlovecake/SATAN LOVES CAKE.sh:31]: Before sourcing custom mod files
```suggestion
```

File: ports/satlovecake/SATAN LOVES CAKE.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+get_controls
+
+# Enable logging
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# Set necessary exports
+export LD_LIBRARY_PATH="/usr/lib:/usr/lib32:$GAMEDIR/lib:$LD_LIBRARY_PATH"
+export GMLOADER_DEPTH_DISABLE=1
+export GMLOADER_SAVEDIR="$GAMEDIR/"
+export GMLOADER_PLATFORM="os_windows"
```
Comment (cdeletre) [inline ports/satlovecake/SATAN LOVES CAKE.sh:30]: Not used by gmloadernext (even if it's the armhf build)
```suggestion
```

File: ports/satlovecake/SATAN LOVES CAKE.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+# Start game with game.droid file in $GAMEDIR
+$GPTOKEYB "gmloadernext" &
+./gmloadernext "$GAMEDIR/game.apk"
+
+# Kill gptokeyb process
+$ESUDO kill -9 $(pidof gptokeyb)
+
+# Clear the console
+printf "\033c" > /dev/tty1
+printf "\033c" > /dev/tty0
```
Comment (cdeletre) [inline ports/satlovecake/SATAN LOVES CAKE.sh:57]:
```suggestion
pm_finish
```

File: ports/satlovecake/SATAN LOVES CAKE.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    echo "No data.win file found in the main folder."
+fi
+
+# Ensure necessary permissions
+$ESUDO chmod 666 /dev/uinput
+$ESUDO chmod +x "$GAMEDIR/gmloadernext"
+
+# Start game with game.droid file in $GAMEDIR
+$GPTOKEYB "gmloadernext" &
+./gmloadernext "$GAMEDIR/game.apk"
```
Comment (cdeletre) [inline ports/satlovecake/SATAN LOVES CAKE.sh:50]: Missing `pm_platform_helper` and gmloadernext doesn't use command line parameters
```suggestion
pm_platform_helper "$GAMEDIR/gmloadernext"
./gmloadernext
```

File: ports/satlovecake/SATAN LOVES CAKE.sh [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+# PortMaster info
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
+GAMEDIR=/$directory/ports/satlovecake
+get_controls
```
Comment (cdeletre) [inline ports/satlovecake/SATAN LOVES CAKE.sh:21]: We usually have this order even if it's not a problem here.
```suggestion
get_controls
GAMEDIR=/$directory/ports/satlovecake
```

**Review by cdeletre — CHANGES_REQUESTED** [review]

File: ports/satlovecake/SATAN LOVES CAKE.sh (hunk 6 above)
Comment (MadShmupper) [reply ports/satlovecake/SATAN LOVES CAKE.sh]: I remade the .sh and uploaded it. I hope this corrects the things you requested fixed.🎩🙂

File: ports/satlovecake/SATAN LOVES CAKE.sh [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+# Rename data.win to game.droid if it exists in the main folder
+if [ -e "$GAMEDIR/data.win" ]; then
+    mv "$GAMEDIR/data.win" "$GAMEDIR/game.droid" || exit 1
+    echo "Renamed data.win to game.droid."
+else
+    echo "No data.win file found in the main folder."
+fi
+
+# Ensure necessary permissions
+$ESUDO chmod 666 /dev/uinput
```
Comment (Cebion) [inline ports/satlovecake/SATAN LOVES CAKE.sh:42]: This is no longer needed.
```suggestion
```

File: ports/satlovecake/SATAN LOVES CAKE.sh [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+# PortMaster info
+source $controlfolder/control.txt
+export PORT_32BIT="Y"
```
Comment (Cebion) [inline ports/satlovecake/SATAN LOVES CAKE.sh:19]: Is this a 32 bit gmloader next? If yes ignore this comment. Else line need to be removed.

File: ports/satlovecake/SATAN LOVES CAKE.sh [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+GAMEDIR=/$directory/ports/satlovecake
+
+# Enable logging
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# Set necessary exports
+export LD_LIBRARY_PATH="/usr/lib:/usr/lib32:$GAMEDIR/lib:$LD_LIBRARY_PATH"
```
Comment (Cebion) [inline ports/satlovecake/SATAN LOVES CAKE.sh:28]: Same here, is this gmloadernext 32 bit?
If not the ld_library_path is wrong.

**Review by Cebion — CHANGES_REQUESTED** [review]
See comments

File: ports/satlovecake/SATAN LOVES CAKE.sh (hunk 7 above)
Comment (MadShmupper) [reply ports/satlovecake/SATAN LOVES CAKE.sh]: Just fixed it🙂

**Review by Slayer366 — APPROVED** [review]

File: ports/satlovecake/SATAN LOVES CAKE.sh (hunk 8 above)
Comment (cdeletre) [reply ports/satlovecake/SATAN LOVES CAKE.sh:16]: yes it's armhf build :)

**Review by cdeletre — APPROVED** [review]

**Review by Cebion — APPROVED** [review]

**Pull request merged**